server/backends/disk_cache/disk_cache.go

```diff
@@ -718,15 +718,18 @@ func (p *partition) contains(ctx context.Context, cacheType interfaces.CacheType
 	// [ActionResult][build.bazel.remote.execution.v2.ActionResult] and will be
 	// for some period of time afterwards. The TTLs of the referenced blobs SHOULD be increased
 	// if necessary and applicable.
-	p.mu.Lock()
-	defer p.mu.Unlock()
+	p.mu.RLock()
 	ok := p.lru.Contains(k.FullPath())
+	isMapped := p.diskIsMapped
+	p.mu.RUnlock()
 
-	if !ok && !p.diskIsMapped {
+	if !ok && !isMapped {
 		// OK if we're here it means the disk contents are still being loaded
 		// into the LRU. But we still need to return an answer! So we'll go
 		// check the FS, and if the file is there we'll add it to the LRU.
+		p.mu.Lock()
 		ok = p.addFileToLRUIfExists(k)
```

Comment:
Are we concerned at all that the state of the var `p.diskIsMapped` may change between when we assign it to `isMapped` and when we actually call `p.addFileToLRUIfExists`?

Reply:
Yes, I did check this -- since addFileToLRUIfExists checks that var again, I think this code is safe?

Reply:
(That said, the test is currently failing so there might be some state change wonkiness here still to debug)

Reply:
slight tweak -- same lift and test passes.